pkg/kubelet/kubelet_pods.go

```diff
@@ -414,6 +414,44 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 		return result, err
 	}
 
+	var (
+		configMaps = make(map[string]*v1.ConfigMap)
+		tmpEnv     = make(map[string]string)
+	)
+	for _, envFrom := range container.EnvFrom {
+		if envFrom.ConfigMapRef != nil {
+			name := envFrom.ConfigMapRef.Name
+			configMap, ok := configMaps[name]
+			if !ok {
+				if kl.kubeClient == nil {
+					return result, fmt.Errorf("Couldn't get configMap %v/%v, no kubeClient defined", pod.Namespace, name)
+				}
+				configMap, err = kl.kubeClient.Core().ConfigMaps(pod.Namespace).Get(name, metav1.GetOptions{})
+
+				if err != nil {
+					return result, err
+				}
+				configMaps[name] = configMap
+			}
+			for k, v := range configMap.Data {
+				if errMsgs := utilvalidation.IsCIdentifier(k); len(errMsgs) != 0 {
```

Comment:
This is the behavior you get if you define invalid env vars as well. The problem here is that we need to do it at run time instead. But this is how the proposal was written.

We can't transform it since we don't know what was invalid. Skipping is even worse since you have no idea why an environment variable went missing.
Currently you will see your container failed to start and a message in the log.